tools/js-tools/git-hooks/pre-push-hook.js

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env node
+
+/* eslint-disable no-console, no-process-exit */
+const { spawnSync } = require( 'child_process' );
+const chalk = require( 'chalk' );
+
+console.log( chalk.green( 'Checking if changelog files are needed. Just a sec...' ) );
+
+/**
+ * Checks if changelog files are required.
+ */
+function checkChangelogFiles() {
+	const needChangelog = spawnSync(
+		'tools/check-changelogger-use.php',
+		[ 'origin/master', 'HEAD' ],
+		{
+			stdio: 'inherit',
+		}
+	);
+
+	// If a changelog file is needed, quit the push.
+	if ( needChangelog.status === 1 ) {
+		process.exitCode = 1;
+	}
+
+	if ( needChangelog.status === 0 ) {
+		console.log( chalk.green( 'Changelog check passed.' ) );
+	}
```

Comment:
Any non-zero status should be propagated to the exit code, so we can simplify it like
```suggestion
	if ( needChangelog.status === 0 ) {
		console.log( chalk.green( 'Changelog check passed.' ) );
	} else {
		process.exitCode = 1;
	}
```

Reply:
Done 👍 051f912